torch/csrc/utils/auto_cuda_event.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+// RAII struct to to record and wait on CUDA events
+
+#ifdef USE_CUDA
+
+#include "cuda.h"
+#include "cuda_runtime.h"
+#include <ATen/ATen.h>
+#include <THC/THC.h>
+
+struct AutoCUDAEvent {
```

Comment:
We usually use `AutoX` to signify a "scope manager" that changes some thread local settings for its duration. This is not what this struct does. Can you call it just `CUDAEvent`?